src/dag-service.js

```diff
@@ -25,33 +25,78 @@ module.exports = class DAGService {
 
   // get retrieves a DAGNode, using the Block Service
   get (multihash, callback) {
-    const isMhash = isIPFS.multihash(multihash)
-    const isPath = isIPFS.path(multihash)
+    this.getMany([multihash], (err, results) => {
+      if (err) {
+        return callback(err)
+      }
 
-    if (!isMhash && !isPath) {
-      return callback(new Error('Invalid Key'))
-    }
+      const key = Object.keys(results)[0]
 
-    if (isMhash) {
-      this.getWith(multihash, callback)
-    }
+      if (results[key].error) {
+        return callback(results[key].error)
+      }
 
-    if (isPath) {
-      const ipfsKey = multihash.replace('/ipfs/', '')
-      this.getWith(ipfsKey, callback)
-    }
+      callback(null, results[key])
+    })
+  }
+
+  getMany (mhs, callback) {
+    let err = false
+    const list = mhs.map((m) => {
+      const isMhash = isIPFS.multihash(m)
+      const isPath = isIPFS.path(m)
+
+      if (!isMhash && !isPath) {
+        err = true
+        callback(new Error('Invalid Key'))
+        return
+      }
+      if (isMhash) return m
+      if (isPath) return m.replace('/ipfs/', '')
+    })
+
+    if (err) return
+
+    this.getWithMany(list, callback)
   }
 
   getWith (key, callback) {
-    const formatted = typeof key === 'string' ? new Buffer(base58.decode(key)) : key
-    this.bs.getBlock(formatted, (err, block) => {
+    this.getWithMany([key], (err, results) => {
       if (err) {
         return callback(err)
       }
 
-      const node = new DAGNode()
-      node.unMarshal(block.data)
-      return callback(null, node)
+      const key = Object.keys(results)[0]
+
+      if (results[key].error) {
+        return callback(results[key].error)
+      }
+
+      callback(null, results[key].block)
+    })
+  }
+
+  getWithMany (keys, callback) {
```

Comment:
Could you document what this method does? Both of

- [ ] here (doc comment)
- [ ] in the readme (api listing)